Source/WebCore/page/writing-tools/WritingToolsController.mm

```diff
@@ -355,35 +355,80 @@
     }
 }
 
+void WritingToolsController::showSelectionForWritingToolsSessionWithID(const WritingTools::Session::ID& sessionID) const
+{
+    RefPtr document = this->document();
+    if (!document) {
+        ASSERT_NOT_REACHED();
+        return;
+    }
+
+    auto it = m_states.find(sessionID);
+    if (it == m_states.end()) {
+        ASSERT_NOT_REACHED();
+        return;
+    }
+
+    auto* state = std::get_if<CompositionState>(&it->value);
```

Comment:
the CompositionState lasts the lifetime of the writing tools controller, since it's just a simple struct type member variable of the controller. This is also in dozens of other places across the file, so I will see if they can all be CheckPtr'd and if so change them all at once in a follow up

Reply:
Sounds good. We should probably deploy smart pointers here to enforce this invariant:

> CompositionState lasts the lifetime of the writing tools controller